### PR TITLE
Add a nil check for CAPI diff for upgrade plan

### PR DESCRIPTION
**Pull request by mitalipaygude**
*Description of changes:*
Add a nil check for CAPI diff for upgrade plan. Customer saw a nil pointer thrown while running the upgrade plan.

CAPI change diff does not have the Components Reports initialized because of empty Flux Change Diff.
*Testing (if applicable):*
- make and manual testing

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/types/upgrades.go [hunk 1]
```diff
@@ -26,6 +26,9 @@ func NewChangeDiff(componentReports ...*ComponentChangeDiff) *ChangeDiff {
 func (c *ChangeDiff) Append(changeDiffs ...*ChangeDiff) {
 	for _, diff := range changeDiffs {
 		if diff != nil {
+			if c == nil {
```
Comment (vivek-koppuru) [inline pkg/types/upgrades.go:29]: How is this possible if `c` is a member method?

Comment (vivek-koppuru) [reply pkg/types/upgrades.go:29]: Nvm after seeing https://stackoverflow.com/questions/42238624/calling-a-method-on-a-nil-struct-pointer-doesnt-panic-why-not

File: pkg/constants/constants.go [hunk 2]
```diff
@@ -54,14 +54,14 @@ const (
...
 	// Provider specific env vars.
-	VSphereUsernameKey = "VSPHERE_USERNAME"
-	VSpherePasswordKey = "VSPHERE_PASSWORD"
-	GovcUsernameKey    = "GOVC_USERNAME"
-	GovcPasswordKey    = "GOVC_PASSWORD"
-	SnowCredentialsKey = "AWS_B64ENCODED_CREDENTIALS"
-	SnowCertsKey       = "AWS_B64ENCODED_CA_BUNDLES"
-	NutanixUsernameKey = "NUTANIX_USER"
-	NutanixPasswordKey = "NUTANIX_PASSWORD"
+	VSphereUsernameKey     = "VSPHERE_USERNAME"
```
Comment (mitalipaygude) [inline pkg/constants/constants.go:57]: gofumpt formatting changes

File: pkg/curatedpackages/curatedpackages.go [hunk 3]
```diff
@@ -62,6 +62,7 @@ func PrintLicense() {
 	fmt.Println(license)
 	fmt.Println(userMsgSeparator)
 }
+
```
Comment (mitalipaygude) [inline pkg/curatedpackages/curatedpackages.go:65]: gofumpt formatting change

**Review by vivek-koppuru — APPROVED** [review]

**Pull request merged**